### PR TITLE
Housekeeping - Gradle, License, Forge Registries

### PR DESCRIPTION
**Pull request by jamierocks**
- Rewrote the Gradle buildscripts
  - Bumped Gradle wrapper to 4.1
  - Removed the MDK junk
  - Updated the MCP mappings to more recent ones
- Fixed license headers
- Produced, using gitignore.io, an improved .gitignore file
- Added a .gitattributes file to ensure Git diffs files correctly
- ~~Move constants to a SmoresConstants class~~
- Add missing newlines
- Use the Forge registries
- Follow Minecraft package layout (blocks -> block, items -> item)
- Ran my IntelliJ formatter

### CODE REVIEW

**Comment by darkevilmac** [comment]
> Move constants to a SmoresConstants class

but why tho

**Comment by jamierocks** [comment]
> but why tho

Simply put, I prefer to separate my constants away where they are on their own.

File: src/main/java/com/elytradev/smores/Smores.java [hunk 1]
```diff
@@ -1,100 +1,170 @@
...
-import net.minecraftforge.fml.common.event.FMLInterModComms;
 import net.minecraftforge.fml.common.event.FMLPostInitializationEvent;
 import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
 import net.minecraftforge.fml.common.registry.GameRegistry;
+import net.minecraftforge.fml.common.registry.IForgeRegistry;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+@Mod(
```
Comment (falkreon) [inline src/main/java/com/elytradev/smores/Smores.java:75]: If you really want to put things where they go:
*  Strings that can be localized go in the lang file. In this case, none of them
*  Strings that specify part of the build configuration, such as the version, go in the build file and get replaced here.
*  Strings that can't be localized, but fall logically within the scope of this object, belong in this object. In this case, modid, name, and guiFactory.

A SmoresConstants class would *only* contain Strings that belong somewhere else.

File: src/main/java/com/elytradev/smores/Smores.java [hunk 2]
```diff
@@ -1,100 +1,170 @@
...
+        guiFactory = "com.elytradev.smores.gui.SmoresGuiFactory")
+@Mod.EventBusSubscriber
+public final class Smores {
 
-@Mod(modid = Smores.SMORES_MOD_ID, name = Smores.SMORES_NAME, version = Smores.SMORES_VERSION, guiFactory = "com.elytradev.smores.gui.SmoresGuiFactory")
-public class Smores {
-    public static final String SMORES_MOD_ID = "smores";
-    public static final String SMORES_NAME = "Smores";
-    public static final String SMORES_VERSION = "1.11.2-1.0.0.0";
+    public static final Logger log = LogManager.getLogger(SmoresConstants.NAME);
```
Comment (falkreon) [inline src/main/java/com/elytradev/smores/Smores.java]: either Smores.class.getName() or Smores.NAME

File: src/main/java/com/elytradev/smores/Smores.java [hunk 3]
```diff
@@ -1,100 +1,170 @@
...
+    }
 
+    private static void registerBlock(IForgeRegistry<Block> registry, Block block) {
+        registry.register(block);
     }
 
-    @Mod.EventHandler
-    public void onIMC(FMLInterModComms.IMCEvent event) {
-        for(FMLInterModComms.IMCMessage message : event.getMessages()) {
-
```
Comment (falkreon) [inline src/main/java/com/elytradev/smores/Smores.java:97]: We're just going to need to put this IMC hook back in for ore voting.

File: src/main/java/com/elytradev/smores/SmoresConstants.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/*
```
Comment (falkreon) [inline src/main/java/com/elytradev/smores/SmoresConstants.java]: no.

File: src/main/java/com/elytradev/smores/item/ItemBase.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import com.elytradev.smores.SmoresConstants;
+import net.minecraft.creativetab.CreativeTabs;
+import net.minecraft.item.Item;
+
+public abstract class ItemBase extends Item {
+
+    protected String name;
+
+    protected ItemBase(String name)
+    {
```
Comment (falkreon) [inline src/main/java/com/elytradev/smores/item/ItemBase.java]: Hanging braces need to be fixed in this class.

File: src/main/java/com/elytradev/smores/Smores.java (hunk 2 above)
Comment (falkreon) [reply src/main/java/com/elytradev/smores/Smores.java]: Also, log is static final. It should be LOG.

**Review by falkreon — CHANGES_REQUESTED** [review]
Overall, a good change. Some of this formatting needed to happen. Some of it really, really didn't. Let's chop this down to what needs to be done.

File: src/main/java/com/elytradev/smores/Smores.java (hunk 1 above)
Comment (jamierocks) [reply src/main/java/com/elytradev/smores/Smores.java:75]: Fair enough, I'll move them back into `Smores`.

File: src/main/java/com/elytradev/smores/Smores.java (hunk 2 above)
Comment (jamierocks) [reply src/main/java/com/elytradev/smores/Smores.java]: Using lowercase for `Logger`s is a convention I have long been accustomed to. Forge also follow this convention - https://github.com/MinecraftForge/MinecraftForge/blob/1.12.x/src/main/java/net/minecraftforge/fml/common/FMLLog.java#L40.

File: src/main/java/com/elytradev/smores/Smores.java (hunk 3 above)
Comment (jamierocks) [reply src/main/java/com/elytradev/smores/Smores.java:97]: The handler did nothing, and would be trivial to rewrite.

File: src/main/java/com/elytradev/smores/item/ItemBase.java (hunk 5 above)
Comment (jamierocks) [reply src/main/java/com/elytradev/smores/item/ItemBase.java]: I did not introduce the hanging braces, but perhaps it falls under the directive of this pr to resolve.

**Review by falkreon — APPROVED** [review]

**Pull request merged**